engine/gamesys/src/gamesys/scripts/script_image.cpp

```diff
@@ -93,6 +103,24 @@ namespace dmScript
     *         local tx = gui.new_texture("image_node", img.width, img.height, img.type, img.buffer)
     *     end)
     * ```
+    *
+    * Load an image from an URL as a buffer and create a texture resource from it:
+    *
+    * ```lua
+    * local imgurl = "http://www.site.com/image.png"
+    * http.request(imgurl, "GET", function(self, id, response)
+    *         local img = image.load(response.response, { buffer = true })
+    *         local tparams = {
+    *             width  = img.width,
+    *             height = img.height,
+    *             type   = resource.TEXTURE_TYPE_2D,
+    *             format = resource.TEXTURE_FORMAT_RGBA }
+    *
+    *         local my_texture_id = resource.create_texture("/my_custom_texture.texturec", tparams, img.buffer)
```

Comment:
One can also imagine the api allowing a `string`, and we'll create a temporary buffer on the fly.
Which might be handy in the future if someone want to downkload a resource, and add it directlty into the resource system (without "loading" it) (e.g. we have a new api which we're using for the Rive extension, where we simply insert new precompiled resources)

Reply:
Another api would be to create a buffer directly from a string?

Reply:
Was an `image.load_buffer()` also considered?

While adding an extra argument to the options table is likely the least invasive change we could make, and also saves code size. But I also think it's a slightly odd api.

And it sets a precedence that we can return different (unrelated) types from the same function, which isn't usually how we code in general.